src/components/ArticlesByTopic.jsx

```diff
@@ -1,36 +1,52 @@
 import React, { useState, useEffect } from "react";
-import { useParams } from "react-router-dom";
-import axios from "axios";
+import { useParams, useSearchParams } from "react-router-dom";
 import ArticleCard from "./ArticleCard";
+import SortControls from "./SortControls";
+import { fetchArticles } from "../utils/api";
 
 const ArticlesByTopic = () => {
   const { topic_slug } = useParams();
   const [articles, setArticles] = useState([]);
   const [loading, setLoading] = useState(true);
   const [error, setError] = useState(null);
+  const [searchParams, setSearchParams] = useSearchParams();
 
   useEffect(() => {
-    axios
-      .get(
-        `https://cwazycodes-nc-news.onrender.com/api/articles?topic=${topic_slug}`
-      )
-      .then((response) => {
-        setArticles(response.data.articles);
+    const sort_by = searchParams.get("sort_by") || "created_at";
+    const order = searchParams.get("order") || "desc";
+
+    fetchArticles(topic_slug, sort_by, order)
+      .then((articlesData) => {
+        setArticles(articlesData);
```

Comment:
Your `ArticlesByTopic` component shares a lot of the same code that is also found in the `ArticleList` component, you could combine these into one component that is used in both situations dynamically